lib/vector-config/src/configurable.rs

```diff
@@ -65,6 +65,14 @@ where
     /// If an error occurs while generating the schema, an error variant will be returned describing
     /// the issue.
     fn generate_schema(gen: &RefCell<SchemaGenerator>) -> Result<SchemaObject, GenerateError>;
+
+    /// Create a new configurable reference table.
+    fn make_ref() -> ConfigurableRef
```

Comment:
Note: I'm not particularly happy about this function name, so if you have a better idea I would welcome hearing it.

Reply:
I'd probably just call it `as_configurable_ref`, which feels a bit more Rust-y, even if the return value isn't _literally_ a reference.